src/misc/programs.cpp

```diff
@@ -39,43 +41,51 @@ Bitu call_program;
 
 /* This registers a file on the virtual drive and creates the correct structure for it*/
 
-static Bit8u exe_block[]={
-	0xbc,0x00,0x04,					//MOV SP,0x400 decrease stack size
-	0xbb,0x40,0x00,					//MOV BX,0x040 for memory resize
-	0xb4,0x4a,						//MOV AH,0x4A	Resize memory block
-	0xcd,0x21,						//INT 0x21
-//pos 12 is callback number
-	0xFE,0x38,0x00,0x00,			//CALLBack number
-	0xb8,0x00,0x4c,					//Mov ax,4c00
-	0xcd,0x21,						//INT 0x21
+constexpr uint8_t exe_block[] = {
+        0xbc, 0x00, 0x04,       // MOV SP,0x400  Decrease stack size
+        0xbb, 0x40, 0x00,       // MOV BX,0x040  For memory resize
+        0xb4, 0x4a,             // MOV AH,0x4A   Resize memory block
+        0xcd, 0x21,             // INT 0x21
+        0xFE, 0x38, 0x00, 0x00, // 12th position is the CALLBack number
+        0xb8, 0x00, 0x4c,       // Mov ax,4c00
+        0xcd, 0x21,             // INT 0x21
 };
 
-#define CB_POS 12
+// COM data constants
+constexpr int callback_pos = 12;
+constexpr int com_data_size = 32;
 
-static std::vector<PROGRAMS_Main*> internal_progs;
+// Persistent program containers
+using com_data_t = std::unique_ptr<uint8_t[]>;
+static std::vector<com_data_t> internal_progs_comdata;
+static std::vector<PROGRAMS_Main *> internal_progs;
 
-void PROGRAMS_MakeFile(char const * const name,PROGRAMS_Main * main) {
-	Bit8u * comdata=(Bit8u *)malloc(32); //MEM LEAK
+void PROGRAMS_MakeFile(const char *name, PROGRAMS_Main *main)
+{
+	com_data_t comdata = std::make_unique<uint8_t[]>(com_data_size);
+	assert(comdata);
 
-	if (comdata == nullptr) {
-		E_Exit("Could not allocate memory for com-file data.");
-	}
```

Comment:
Here's where we replace malloc with a smart pointer.
`make_unique` with throw a `bad_alloc` exception if it fails, so we can drop error-checking and `E_Exit()`.

Reply:
any desire to go further with the C++, such as:
```
constexpr std::array<uint8_t, 19> exe_block = ...
```
and 
```
	std::copy(exe_block.begin(), exe_block.end(), comdata.get());
```

and `exe_block.size()` vs. `sizeof(exe_block)`, etc.?